frontend/src/lib/components/InsightSubscription/insightSubscriptionLogic.ts

```diff
@@ -0,0 +1,81 @@
+import { afterMount, connect, kea, key, path, props } from 'kea'
+import { InsightShortId, SubscriptionType } from '~/types'
+
+import api from 'lib/api'
+import { loaders } from 'kea-loaders'
+import { forms } from 'kea-forms'
+
+import type { insightSubscriptionLogicType } from './insightSubscriptionLogicType'
+import { isEmail } from 'lib/utils'
+import { dayjs } from 'lib/dayjs'
+import { lemonToast } from '../lemonToast'
+import { insightLogic } from 'scenes/insights/insightLogic'
+import { router } from 'kea-router'
+import { urls } from 'scenes/urls'
+import { insightSubscriptionsLogic } from './insightSubscriptionsLogic'
+
+const NEW_SUBSCRIPTION: Partial<SubscriptionType> = {
+    frequency: 'weekly',
+    interval: 1,
+    start_date: dayjs().hour(9).minute(0).second(0).toISOString(),
+    title: 'New Subscription',
+    target_type: 'email',
+}
+
+export interface InsightSubscriptionLogicProps {
+    id: number | 'new'
+    insightShortId: InsightShortId
+}
+export const insightSubscriptionLogic = kea<insightSubscriptionLogicType>([
```

Comment:
flyby without proper review...

we should get tests around the logics before they (maybe possibly in the future) get more complicated or it is painful to backfill

Reply:
I love that one of my first comments on a PR of yours was - "maybe this should have tests" and now I'm getting the same treatment. You're keeping me an honest man 😉

Totally agreed. I've generally been quite lax on the frontend tests which i should try and rectify I think

Reply:
![spiderhogs-54f1bf6e8ebf31791e4c5c1813b19f2a](https://user-images.githubusercontent.com/984817/172697363-7c309662-0b17-451b-b1cc-044b495dda63.png)